model/flow/address.go

```diff
@@ -273,6 +273,9 @@ const invalidCodeTransientNetwork = uint64(0x1cb159857af02018)
 // invalidCodeSandboxNetwork is the invalid codeword used for Sandbox network.
 const invalidCodeSandboxNetwork = uint64(0x1035ce4eff92ae01)
 
+// invalidCodePreviewNetwork is the invalid codeword used for Preview networks.
```

Comment:
>Still need [invalidCodePreviewNetwork](https://github.com/onflow/flow-go/compare/feature/stable-cadence...kan/preview-net-chain-id?expand=1#diff-ea335e5bacc5828fbe8db5edd37db9f63bb4d363e545f185a52b0f89a6f3a852R277) if we actually do need a brand new chain ID. Typically, this is something i'd get @tarakby's help with.

Can we add comment how this codeword is produced? or is there a test function to produce or validate this value?

Reply:
a brief explanation for all the invalid code words is given [here](https://github.com/onflow/flow-go/blob/kan/preview-net-chain-id/model/flow/address.go#L264-L265). All the invalid code words are validated by this [test](https://github.com/onflow/flow-go/blob/kan/preview-net-chain-id/model/flow/address_test.go#L260) : all address spaces created are disjoint (each two don't intersect)

Reply:
added more comments here: https://github.com/onflow/flow-go/pull/5396/commits/f627367d6c0f2b9f6b5979a6b93d91459269794c. 
Understanding all the details require going through the other comments in the implementation, and reading the basics about Hamming codes.

Reply:
From my layman understanding, I think for generation it is something like this: 
```
while
    r = rand_uint64() 
    if r != 0 and not isValidCodeWord(r):
      break
```

Reply:
Correct, though `isValidCodeWord(r)` should be applied to all existing networks (Mainnet and all others). I also choose `r` with a hamming weight close to 32 so that we distance the test space from the Mainnet space

Reply:
@tarakby off topic but I was wondering that recently, is it possible to write a function that suggests closest address ( when user writes the address wrong ) something like `did you mean xxxxxx ?`

Reply:
@tarakby Could you maybe add this "layman understanding" to the comment? It helped me understand what is happening there :-)

Reply:
@bluesign 
> is it possible to write a function that suggests closest address ( when user writes the address wrong ) 

Good question! the scheme was designed only for error detection (up to 6 bits) and not error correction.  However I speculate the scheme can also provide up to 3 bits of error correction (no proof, only gut feeling), which means we can guess the original address correctly if we believe up to 3 bits only got flipped. 
If more bits were flipped, we can still make a guess, but the guess will be wrong (different than the original address), which is dangerous from a blockchain user perspective (imagine sending tokens to the wrong address because it was suggested by an interface). I think it's safer to only check the address and reject any invalid address, to force the user to double check.

Reply:
> Could you maybe add this "layman understanding" to the comment? It helped me understand what is happening there

makes sense, and it's useful for future extensions of the code. I'll add it in another PR